ext/src/main/java/com/kickstarter/jruby/Telekinesis.java

```diff
@@ -45,11 +48,16 @@ public static Worker newWorker(final KinesisClientLibConfiguration config,
             }
         };
 
-        return new Worker.Builder()
+        Worker.Builder builder = new Worker.Builder()
                 .recordProcessorFactory(v2Factory)
                 .config(config)
-                .execService(executor) // NOTE: .execService(null) is a no-op
-                .build();
+                .execService(executor); // NOTE: .execService(null) is a no-op
+
+        if (dynamoClient != null) {
+            builder.dynamoDBClient(dynamoClient);
+        }
```

Comment:
`.dynamoDBClient(null)` is the default value, so it's safe to not do this `if` check.